src/OrchardCore.Modules/OrchardCore.Search/SearchAreaControllerRouteMapper.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Routing;
+using OrchardCore.Entities;
+using OrchardCore.Mvc.Routing;
+using OrchardCore.Search.Model;
+using OrchardCore.Settings;
+
+namespace OrchardCore.Search
+{
+    public class SearchAreaControllerRouteMapper : IAreaControllerRouteMapper
+    {
+        private readonly ISiteService _site;
+
+        public int Order => -1000;
+
+        public SearchAreaControllerRouteMapper(ISiteService site)
+        {
+            _site = site;
+        }
+
+        public bool TryMapAreaControllerRoute(IEndpointRouteBuilder routes, ControllerActionDescriptor descriptor)
+        {
+            var settings = _site.GetSiteSettingsAsync()
+                .GetAwaiter().GetResult()
+                .As<SearchSettings>();
+
+            if (!String.IsNullOrEmpty(settings.SearchProvider))
+            {
+                routes.MapAreaControllerRoute(
```

Comment:
You also need to have the Search provider settings set.

Reply:
Since that dropdown under /Admin/Settings/search always shows one of the providers selected (not like "Unconfigured"), this isn't clear. However, that shouldn't really be necessary to save either: Let's use the provider that appears there at the top by default. That way, if you enable just Elastic or Lucene, it'll just work. If you enable both, you get a default, and if you disable one of them next, the remaining one will also keep working.

Reply:
I see what you mean. Yes, that makes sense.

Reply:
done